sourced/ml/utils/engine.py

```diff
@@ -105,12 +105,16 @@ def create_engine(session_name, repositories,
     # TODO(vmarkovtsev): figure out why is this option needed
     config.append("spark.tech.sourced.engine.cleanup.skip=true")
     packages.append("tech.sourced:engine:" + engine)
+    memory_conf = []
     if memory:
         memory = memory.split(",")
-        memory.append("spark.executor.memory=" + memory[0])
-        memory.append("spark.driver.memory=" + memory[1])
-        memory.append("spark.driver.maxResultSize=" + memory[2])
-    session = create_spark(session_name, config=config, packages=packages, memory=memory,
+        err = "Expected 3 memory parameters but got {}. " \
+              "Please check --help to see how -m/--memory should be used."
+        assert len(memory) == 3, err.format(len(memory))
```

Comment:
I think it is better to use
```
if len(memory) == 3:
    raise ValueError(err)
```

`assert ` more for debug purposes.

Reply:
`raise` can be useful if you want to handle it somehow. I don't think wrong parameters to script should be handled in this situation because it may lead to some unexpected behaviour for future users. And you right - the purpose was to make it easy to understand the error by users.